scvi/model/utils/_minification.py

```diff
@@ -21,12 +22,18 @@ def get_minified_adata_scrna(
         Original adata, of which we to create a minified version.
     minified_data_type
         How to minify the data.
+    keep_count_data
+        Whether to keep the count data and only store additionally the latent posterior.
     """
-    if minified_data_type != ADATA_MINIFY_TYPE.LATENT_POSTERIOR:
+    if not ADATA_MINIFY_TYPE.__contains__(minified_data_type):
         raise NotImplementedError(f"Unknown MinifiedDataType: {minified_data_type}")
 
-    all_zeros = csr_matrix(adata.X.shape)
-    layers = {layer: all_zeros.copy() for layer in adata.layers}
+    if not keep_count_data:
+        all_zeros = csr_matrix(adata.X.shape)
+        layers = {layer: all_zeros.copy() for layer in adata.layers}
+    else:
+        all_zeros = adata.X
+        layers = adata.layers
```

Comment:
It may also make sense to only copy the count layer registered with the manager so we don't have to carry around the unused layers as part of the minified data.